### ISSUE TITLE
TaxJar creates a WP User?

### ISSUE THREAD

**Issue by bhubbard** [taxjar/taxjar-woocommerce-plugin#115]
So going thru the code a bit on the plugin, I'm a bit confused? Why does TaxJar create a WordPress User for the API?

**Comment by dallendalton** [comment]
@bhubbard - So the TaxJar plugin does not just calculate tax for orders placed on the store. It can also (if the setting is enabled) sync completed orders to the TaxJar app and use that data to calculate which states a company has nexus in, how much they have collected and need to remit, and for an additional cost can automatically file sales tax for the company.

Currently, the TaxJar app accesses the necessary transaction data through the WooCommerce REST API of the store. This is why it has to create a WP User. In older WooCommerce versions, the REST API tokens are tied to a specific user account, so in order to create the tokens we have to first create a user. In newer versions (after 2.4), the API tokens can be created without creating a new user, and the TaxJar plugin will not create a new user.

In summary, it is a feature that is in place to ensure backwards compatibility for older versions of WooCommerce. The only time the plugin will create a user is if you have download orders enabled and if your WooCommerce version is older than 2.4.

Just as a note, our next major version will be removing a lot of this code for creating users and API tokens. We are moving to a new system that will instead push orders in near real time from the WooCommerce store to the TaxJar app through the TaxJar API. This will allow us to introduce some features to the plugin. Thanks!

**Comment by bhubbard** [comment]
Thanks @dallendalton that makes sense.
